lua/ttt2/extensions/draw.lua

```diff
@@ -0,0 +1,436 @@
+-- @param number yalign The alignment of the Y coordinate using
+-- <a href="https://wiki.garrysmod.com/page/Enums/TEXT_ALIGN">TEXT_ALIGN_Enums</a>.
+-- @param boolean shadow whether there should be a shadow of the text
+-- @param number scale scale (float number)
+-- @module draw
+-- @realm client
+function draw.AdvancedText(text, font, x, y, color, xalign, yalign, shadow, scale)
+	local scaleModifier = 1.0
+	local t_font = fonts.GetFont(font)
+
+	if t_font then
+		scaleModifier = fonts.GetScaleModifier(scale)
+		font = t_font[scaleModifier]
+		scale = scale / scaleModifier
+	end
+
+	local scaled = isvector(scale) or scale ~= 1.0
+	local mat
+
+	if scaled then
+		local hw = ScrW() * 0.5
+		local hh = ScrH() * 0.5
+
+		mat = Matrix()
+		mat:Translate(Vector(x, y))
+		mat:Scale(isvector(scale) and scale or Vector(scale, scale, scale))
+		mat:Translate(-Vector(hw, hh))
+
+		render.PushFilterMag(TEXFILTER.LINEAR)
+		render.PushFilterMin(TEXFILTER.LINEAR)
+
+		cam.PushModelMatrix(mat)
+
+		x = hw
+		y = hh
+	end
+
+	if shadow then
+		drawShadowedText(text, font, x, y, color, xalign, yalign, scaleModifier)
+	else
+		drawSimpleText(text, font, x, y, color, xalign, yalign)
+	end
+
+	if scaled then
+		cam.PopModelMatrix(mat)
+
+		render.PopFilterMag()
+		render.PopFilterMin()
+	end
+end
+
+---
+-- Returns a list of lines to wrap the text matching the given width
+-- @param string text
+-- @param number width
+-- @param string font
+-- @param number scale
+-- @return table
+-- @realm client
+function draw.GetWrappedText(text, width, font, scale)
+	-- Oh joy, I get to write my own wrapping function. Thanks Lua!
+	-- Splits a string into a table of strings that are under the given width.
+
+	scale = scale or 1.0
+	width = width / scale
+
+	if not text then
+		return {}, 0, 0
+	end
+
+	surface.SetFont(font or "DefaultBold")
+
+	-- Any wrapping required?
+	local w, h = surface.GetTextSize(text)
+
+	if w <= width then
+		return {text}, w, h -- Nope, but wrap in table for uniformity
+	end
+
+	local words = string.Explode(" ", text) -- No spaces means you're screwed
+	local lines = {""}
+
+	for i = 1, #words do
+		local wrd = words[i]
+
+		if i == 1 then
+			-- add the first word whether or not it matches the size to prevent
+			-- weird empty first lines and ' ' in front of the first line
+			lines[1] = wrd
+
+			continue
+		end
+
+		local lns = #lines
+		local added = lines[lns] .. " " .. wrd
+
+		w = surface.GetTextSize(added)
+
+		if w > width then
+			lines[lns + 1] = wrd -- New line needed
+		else
+			lines[lns] = added -- Safe to tack it on
+		end
+	end
+
+	local lns = #lines
+
+	-- get length of longest line
+	local length = 0
+
+	for i = 1, lns do
+		local line_w = surface.GetTextSize(lines[i])
+
+		if line_w > length then
+			length = line_w
+		end
+	end
+
+	-- get height of lines
+	local _, line_h = surface.GetTextSize(text)
+
+	return lines, length * scale, line_h * lns * scale
```

Comment:
what are these random comments? please remove them or write them a way they are actually useful.
in-code comments should not explain what is done but rather why it is done